devito/passes/clusters/aliases.py

```diff
@@ -1191,6 +1196,16 @@ def free_symbols(self):
     def naliases(self):
         return len(self.aliaseds)
 
+    @property
+    def is_frame(self):
+        """
+        An Alias is said to be a "frame" if the `distances` of the `aliaseds`
+        expressions from the `pivot` form a frame in the linear algebra sense.
+        In essence, any two distances are either orthogonal or one is a multiple
+        of the other by a scalar constant (i.e., a "redundant" vector w.r.t. a basis).
+        """
+        return all(len([e for e in i if e != 0]) <= 1 for i in self.distances)
```

Comment:
Not sure how this match the def

Reply:
a distance is a vector in Rn

Reply:
I know that, but this expression still doesn't really make sense to me w.r.t being  a frame (maybe not the best time to think about it)

Reply:
no?

the inner 

```
len([e for e in i if e != 0])
```

tells you how many non-zero entries in an Rn vector. And the inner condition is s.t. it gives True iff at most one element != 0. The outer condition imposes such condition to all Rn vectors in `self.distances`. Hence it is True if for example `(0, 0, 0, 0)`, `(0, 1, 0, 0)`, `(0, 2, 0, 0)`, `(0, 0, 1, 0)`. And here the Rn vectors are either pairwise orthogonal or scaled. No?